### PR TITLE
[knx] dynamic channel-types

### LINKED ISSUE #15065: [knx] Console commands do not use UoM
Looking at #15049 there seems to be an issue sending data with UoM from the openHAB console to a KNX installation.

We have defined in item with DPTs 7.007 (time in hours) with textual config. Here are the snippets from ```knx.things``` and ```default.items```:
```        
Type number: t_t_h "t std" [ ga="7.007:10/0/7" ]

Number:Time time_h "T [%.1f h]" {channel="knx:device:ip:test:t_t_h"}
```

Receiving data properly respects UoM and converts data to seconds, which is the representation of time on OH.

Using the OH console to send data, however:

```
openhab> openhab:send time_h "7 s"
Command has been sent successfully.
openhab> openhab:items list time_h
time_h (Type=NumberItem, State=7 s, Label=T, Category=null)
openhab> openhab:send time_h "2 h"
Command has been sent successfully.
openhab> openhab:items list time_h
time_h (Type=NumberItem, State=7200 s, Label=T, Category=null)
openhab> openhab:send time_h "5 kg"
Command has been sent successfully.
openhab> openhab:items list time_h
time_h (Type=NumberItem, State=7200 s, Label=T, Category=null)
openhab> openhab:send time_h 1
Command has been sent successfully.
openhab> openhab:items list time_h
time_h (Type=NumberItem, State=1 s, Label=T, Category=null)

```
The internal state is properly updated for h and s; the wrong unit "kg" is ignored. The last command without unit uses default for time (s).

But the knx addon sends out the data in all 4 cases (even the one with wrong unit). It used the plain number without any conversion. See the bus log:
```
A_Group.write 00 07
A_Group.write 00 02
A_Group.write 00 05
A_Group.write 00 01
```

## Expected Behavior

Data converted according to UoM, data on the bus is in hours.

## Steps to Reproduce (for Bugs)

see above

## Context

UoM

## Your Environment

* Version used: (e.g., openHAB and add-on versions)
* Environment name and version (e.g. Chrome 76, Java 8, Node.js 12.9, ...):
* Operating System and version (desktop or mobile, Windows 10, Raspbian Buster, ...):

 4.0.0-SNAPSHOT - Build 3487, custom build of KNX addon, based on current main branch

### PR DESCRIPTION
**Pull request by J-N-K**
Fixes #15065 

@holgerfriedrich WDYT?

### CODE REVIEW

**Comment by holgerfriedrich** [comment]
Thanks, @J-N-K!!! Interesting solution - I will give it a try later today.

Do you already have an idea how many bindings will need a similar fix?

**Comment by J-N-K** [comment]
I guess MQTT and HTTP, maybe modbus, not sure how the channels are created there. I don't know of other bindings that have generic channels and UoM support.

**Comment by holgerfriedrich** [comment]
@J-N-K
I just quickly checked by use-case DPT 7.007: seems to work for units which can be converted properly and for plain numbers.

A bit odd, however is the fact that I can still put "kg" as a unit and the plain number is sent to the bus. I think we should behave as core does and ignore this message (maybe log it, bit do not send....)

It has also an effect on #15066: DPT properly changes after updating the config file. However, this is likely to work only for items you modify.

**Comment by J-N-K** [comment]
I checked with different DPTs in 9.xxx.

**Comment by J-N-K** [comment]
The other issue probably is a core bug.

**Comment by holgerfriedrich** [comment]
Regarding the handling of incompatible units:

```
openhab> openhab:send time_h "9 kg"

11:00:57.415 [INFO ] [openhab.event.ItemCommandEvent       ] - Item 'time_h' received command 9 kg
11:00:57.420 [INFO ] [openhab.event.ItemStatePredictedEvent] - Item 'time_h' predicted to become 9 kg
11:00:57.421 [WARN ] [openhab.core.library.items.NumberItem] - Failed to update item 'time_h' because '9 kg' could not be converted to the item unit 's'
11:00:57.429 [TRACE] [x.internal.handler.DeviceThingHandler] - Handling command '9 kg' for channel 'knx:device:ip:hsb_test:t_t_h'
11:00:57.432 [TRACE] [nding.knx.internal.channel.KNXChannel] - getCommandSpec checking keys '[ga]' for command '9 kg' (class org.openhab.core.library.types.QuantityType)
11:00:57.433 [TRACE] [nding.knx.internal.channel.KNXChannel] - getCommandSpec key 'ga' has expectedTypeClass '[class org.openhab.core.library.types.QuantityType, class org.openhab.core.library.types.DecimalType]', matching command '9 kg' and dpt '7.007'
11:00:57.434 [TRACE] [knx.internal.client.AbstractKNXClient] - writeToKNX groupAddress '10/0/7', commandSpec 'org.openhab.binding.knx.internal.channel.WriteSpecImpl@1fcef44'
11:00:57.436 [WARN ] [binding.knx.internal.dpt.ValueEncoder] - Could not convert 9 kg to unit h, stripping unit only. Check your configuration.
11:00:57.438 [TRACE] [knx.internal.client.AbstractKNXClient] - sendToKNX mappedValue: '9' groupAddress: '10/0/7'
11:00:57.462 [DEBUG] [knx.internal.client.AbstractKNXClient] - Wrote value '9 kg' to datapoint 'command DP 10/0/7 'knx:ip:ip', DPT 7.007, low priority' (0. attempt).
```

This is probably due to the decision that ValueEncoder strips the unit instead of dropping the frame.
But with the warning in the log, this should be fine for now.

**Review by holgerfriedrich — APPROVED** [review]
LGTM!

**Review by jlaur — APPROVED** [review]
LGTM

**Comment by jlaur** [comment]
@J-N-K - can you rephrase the PR title to be more release notes-oriented? 🙂
